adb/adb_protocol.py

```diff
@@ -351,8 +351,12 @@ def Open(cls, usb, destination, timeout_ms=None):
       raise InvalidResponseError(
           'Expected the local_id to be %s, got %s' % (local_id, their_local_id))
     if cmd == b'CLSE':
+      # Some devices seem to be sending CLSE once more after a request, this *should* handle it
+      cmd, remote_id, their_local_id, _ = cls.Read(usb, [b'CLSE', b'OKAY'],
```

Comment:
Is the extra `CLSE` only sent after a bad `OPEN`? I thought it was anytime a `CLSE` was sent?

Reply:
I tried having another read after a CLSE and before the next OPEN, but that caused a crash (failure to unpack with a response containing ‘<61’)